packages/PackageInstaller/src/com/android/packageinstaller/InstallStart.java

```diff
@@ -152,7 +153,15 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
             mAbortInstall = true;
         }
 
-        checkDevicePolicyRestrictions(isTrustedSource);
+        // When DPC disabled user app installation, it restricts the play store aswell
+        // We work arround this by skipping devicePolicyChecks for play store
+        // This needs to be done most likely globally, as main profile may also be restricted
+
+        if (callingPackage.equals(PackageId.PLAY_STORE_NAME)) {
```

Comment:
alternative is to make play store a trusted source but I think that's not a good idea

Reply:
The other case where this might apply is a org owned device, it may also have device policy restrictions on unknown sources.

We can also test for this case properly.

Do you have any suggestions on how to improve this?

REQUEST_INSTALL_PACKAGES is already checked before I think, so this is not bypassed.

Reply:
One idea is `checkDevicePolicyRestrictions(isTrustedSource || callingPackage.equals(PackageId.PLAY_STORE_NAME));`

That way we skip DISALLOW_INSTALL_UNKNOWN_SOURCES check for play, but not sure if that's enough

Reply:
Other idea would to add toogle to GmsCompat for this behaviour. Would be more complicated though. (Installing package for all users would need to check toggle value on all users)

Reply:
Ok. I pm'd him.

For device policy checks: This also works

```java
 checkDevicePolicyRestrictions(isTrustedSource || callingPackage.equals(PackageId.PLAY_STORE_NAME));
````
This would be less invasive then currently skipping everything related to device policy and doesn't require making play a trusted source.
Unless you have something better.

Reply:
> Installing via Apps/App Store would be better, it actually handles the INSTALL_PACKAGES permission, and can be prompted upon profile creation. 

So you mean instead of installExistingPackagesAsUser I would go send intent to app store and then it would install play services during profile creation?

Reply:
> It's one of the possible approach to be taken rather than putting bypasses for sandboxed Play apps just for it to be installed during the profile creation.

This is not during profile creation, that's already handled by maybeInstallPlay in devicepolicymanagerservice.

The changes to installstart are because of Device policy blocking installs from play otherwise. Because REQUEST_INSTALL_PACKAGES permission is not enough when DISALLOW_INSTALL_UNKNOWN_SOURCES is set (as is common nowadays with work profiles). Because play is not a system app this restriction would otherwise cause "Blocked by device admin" when installing apps from play.

Reply:
Replaced by permission USER_TRUSTED_SOURCE